### ISSUE TITLE
💅 noUselessLoneBlockStatements false positive when using try/catch

### ISSUE THREAD

**Issue by hougesen** [biomejs/biome#783]
### Environment information

```bash
CLI:
  Version:                      1.3.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "rio"
  JS_RUNTIME_VERSION:           "v16.20.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/8.19.4"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           true
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

noUselessLoneBlockStatements

### Playground link

https://biomejs.dev/playground/?indentStyle=space&quoteStyle=single&trailingComma=none&lintRules=all&code=ZgB1AG4AYwB0AGkAbwBuACAAdgBhAGwAaQBkAGEAdABlAE8AcgBUAGgAcgBvAHcAKABkAGEAdABhADoAIAB1AG4AawBuAG8AdwBuACkAIAB7AAoAIAAgAC8ALwAgAEQAbwAgAHMAbwBtAGUAdABoAGkAbgBnACAAdwBpAHQAaAAgAGQAYQB0AGEACgAgACAAaQBmACAAKABNAGEAdABoAC4AcgBhAG4AZABvAG0AKAApACAAPgAgADAALgA3ADUAKQAgAHsACgAgACAAIAAgAHIAZQB0AHUAcgBuACAATQBhAHQAaAAuAHIAYQBuAGQAbwBtACgAKQAgAD4AIAAwAC4ANQA7AAoAIAAgAH0ACgAKACAAIAAvAC8AIABvAHIAIAB0AGgAcgBvAHcAIABpAHQACgAgACAAdABoAHIAbwB3ACAAbgBlAHcAIABFAHIAcgBvAHIAKABkAGEAdABhACkAOwAKAH0ACgAKAGYAdQBuAGMAdABpAG8AbgAgAGgAYQBuAGQAbABlAEUAcgByAG8AcgAoAF8AZQByAHIAbwByADoAIAB1AG4AawBuAG8AdwBuACkAIAB7AAoAIAAgAC8ALwAgAC4ALgAuAAoAfQAKAGYAdQBuAGMAdABpAG8AbgAgAGgAYQBuAGQAbABlAE4AbwB0AFYAYQBsAGkAZAAoACkAIAB7AAoAIAAgAC8ALwAgAC4ALgAuAAoAfQAKAAoAZgB1AG4AYwB0AGkAbwBuACAAZABvAFMAbwBtAGUAdABoAGkAbgBnAEgAZQByAGUAKAApACAAewAKACAAIAAvAC8AIAAuAC4ALgAKAH0ACgAKAGUAeABwAG8AcgB0ACAAZgB1AG4AYwB0AGkAbwBuACAAdgBhAGwAaQBkAGEAdABlAFMAdAB1AGYAZgAoACkAIAB7AAoAIAAgAGMAbwBuAHMAdAAgAGQAYQB0AGEAIAA9ACAAewB9ADsACgAKACAAIAB0AHIAeQAgAHsACgAgACAAIAAgAGMAbwBuAHMAdAAgAHYAYQBsAGkAZABhAHQAZQBkACAAPQAgAHYAYQBsAGkAZABhAHQAZQBPAHIAVABoAHIAbwB3ACgAZABhAHQAYQApADsACgAKACAAIAAgACAAaQBmACAAKAB2AGEAbABpAGQAYQB0AGUAZAAgAD0APQA9ACAAZgBhAGwAcwBlACkAIAB7AAoAIAAgACAAIAAgACAAaABhAG4AZABsAGUATgBvAHQAVgBhAGwAaQBkACgAKQA7AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AOwAKACAAIAAgACAAfQAKAAoAIAAgACAAIAAvAC8AIABkAG8AIABzAG8AbQBlACAAbQBvAHIAZQAgAHMAdAB1AGYAZgAKACAAIAAgACAAZABhAHQAYQAuAHYAYQBsAGkAZAAgAD0AIAB0AHIAdQBlADsACgAgACAAfQAgAGMAYQB0AGMAaAAgACgAZQByAHIAbwByACkAIAB7AAoAIAAgACAAIABoAGEAbgBkAGwAZQBFAHIAcgBvAHIAKABlAHIAcgBvAHIAKQA7AAoAIAAgACAAIAByAGUAdAB1AHIAbgA7AAoAIAAgAH0ACgAKACAAIABkAG8AUwBvAG0AZQB0AGgAaQBuAGcASABlAHIAZQAoACkAOwAKAAoAIAAgAHIAZQB0AHUAcgBuACAAZABhAHQAYQA7AAoAfQAKAA%3D%3D

### Expected result

The `noUselessLoneBlockStatements` rule should not trigger for try/catch. 

![image](https://github.com/biomejs/biome/assets/56034786/c8e93097-f691-4fba-a5f1-91c34afe179e)

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by TaKO8Ki** [comment]
Can I work on this?

**Comment by Conaclos** [comment]
> Can I work on this?

You are welcome :) I assigned the issue to you :)
